progressbar_printer.go

```diff
@@ -61,101 +63,107 @@ type ProgressbarPrinter struct {
 }
 
 // WithTitle sets the name of the ProgressbarPrinter.
-func (p ProgressbarPrinter) WithTitle(name string) *ProgressbarPrinter {
+func (p *ProgressbarPrinter) WithTitle(name string) *ProgressbarPrinter {
 	p.Title = name
-	return &p
+	return p
 }
 
 // WithMaxWidth sets the maximum width of the ProgressbarPrinter.
 // If the terminal is smaller than the given width, the terminal width will be used instead.
 // If the width is set to zero, or below, the terminal width will be used.
-func (p ProgressbarPrinter) WithMaxWidth(maxWidth int) *ProgressbarPrinter {
+func (p *ProgressbarPrinter) WithMaxWidth(maxWidth int) *ProgressbarPrinter {
```

Comment:
We may not use a pointer receiver. That way, you would overwrite the values of the "forked" printer.
For example: `pb, _ := pterm.DefaultProgressbar.WithMaxWidth(...)`, would change the `MaxWidth` of the `DefaultProgressbar` then. All `WithXxx` methods must use a non-pointer receiver.